### PR TITLE
Switch to GitHub Actions CI

### PR DESCRIPTION
**Pull request by alyst**
Switches from Travis to GitHub Actions CI and updates *ParserCombinator.jl* depenency ~~(requires andrewcooke/ParserCombinator.jl#33 to be merged and the new 2.1.0 version released)~~.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+DelimitedFiles = "8bb1440f-4735-579b-a4ab-409b98df4dab"
+LightGraphs = "093fc24a-ae57-5d10-9952-331d41423f4d"
+Requires = "ae029012-a4dd-5104-9daa-d747884805df"
+SimpleTraits = "699a6c99-e7fa-54fc-8d76-47d257e15c1d"
+
+[compat]
+EzXML = "≥ 0.9.0"
+LightGraphs = "≥ 1.3.0"
+ParserCombinator = "≥ 2.0.0"
+julia = "≥ 0.7.0"
```
Comment (simonschoelly) [inline Project.toml]: Should we stilld target v0.7?

Comment (sbromberger) [reply Project.toml]: probably not.

**Comment by sbromberger** [comment]
Hi. We don’t use TagBot on any JuliaGraphs projects since it requires write access to the repo.

**Comment by alyst** [comment]
> We don’t use TagBot on any JuliaGraphs projects since it requires write access to the repo.

Sure, I can remove it, but is it a problem to allow write access?

The tests are failing because andrewcooke/ParserCombinator.jl#33 is still not merged (it looks like the package development is hibernated).
I'm not sure what to do about that.

**Comment by sbromberger** [comment]
> Sure, I can remove it, but is it a problem to allow write access?

From my perspective it's a huge problem. It means that if TagBot gets compromised, all repos that depend on TagBot are vulnerable to code manipulation. I don't want JuliaGraphs to be part of that exposure.

It's easy enough to tag manually. TagBot saves you two steps and introduces unknown risk.

**Comment by alyst** [comment]
> It means that if TagBot gets compromised, all repos that depend on TagBot are vulnerable to code manipulation.

Ah, actually *TagBot app* and *TagBot GitHub action* are the two different things. The former one is [deprecated](https://discourse.julialang.org/t/ann-the-tagbot-github-app-is-deprecated-in-favour-of-the-tagbot-github-action/34344), the [latter one](https://github.com/marketplace/actions/julia-tagbot) doesn't require generating an access token that could be compromised and used for something else. And, to be super-safe, you can also clone *JuliaRegistries/TagBot* and configure the action to use your clone.
Of course, everything could be done manually, but keeping the version in git tags and in *Project.toml* in sync + autogenerating changelog draft is a nice thing.

**Comment by sbromberger** [comment]
Both require you to trust code that is running from someone else' repository to not do anything with the write access you're giving it to your repository. That's a bit of a problem.

**Comment by alyst** [comment]
> but then someone has to be in charge of updating and auditing

Well, *TagBot* is in *JuliaRegistries*, so it's a part of Julia ecosystem. As long as no 3rd party got unauthorized access to *TagBot* repo, I'd rather trust the authors.
So it boils down to fast-forwarding the fork and updating the `@tag` (or SHA-1 `@commithash`, to be more secure -- then you probably don't need to fork) line when the *TagBot* authors announce it's necessary to upgrade.
Otherwise I wonder whether the new releases of Julia itself should be audited as well.

**Comment by sbromberger** [comment]
> Otherwise I wonder whether the new releases of Julia itself should be audited as well.

They are, at least in some environments.

**Comment by alyst** [comment]
Finally, ParserCombinator v2.1.0 is released, which unblocks merging this PR and releasing the new GraphIO version.
I tried to initiate CI by opening and closing, but it seems that has to be enabled by the repository maintainers (cc @sbromberger).

**Pull request closed without merging**